### ISSUE TITLE
Enforcing recommended attributes on widgets

### ISSUE THREAD

**Issue by ctaymor** [GoogleChrome/accessibility-developer-tools#228]
We'd love to add audits for the recommended relationships between components with aria-roles. For instance, we are looking at a tabs widget. The [w3c guidelines for the role tabpanel](http://www.w3.org/TR/wai-aria/roles#tabpanel) state it should be associated with a tab, with either the tab having an aria-controls or the tabpanel having an aria-labelledby.

This seems to be a slightly unique case, looking at the [w3c documentation on aria roles](http://www.w3.org/TR/wai-aria/roles), but here are a few similar-ish cases.
* [alertdialog should have an aria-describedby](http://www.w3.org/WAI/PF/aria/roles#alertdialog)
* [dialog should have a label through aria-label or aria-labelledby](http://www.w3.org/WAI/PF/aria/roles#dialog)
* [region should have an aria-labelledby](http://www.w3.org/WAI/PF/aria/roles#region)

Would you be interested in this as a tab-specific rule (since it's a slightly unusual case)?

The Pivotal UI Team
@atomanyih @ctaymor @d-reinhold @kennyw12 @matt-royal @nicw @stubbornella
